index.bs

```diff
@@ -913,10 +927,29 @@ When the {{MLContext/[[contextType]]}} is set to [=context type/default=] with t
 
 <details open algorithm>
   <summary>
-    To <dfn>validate buffer with descriptor</dfn> given {{ArrayBufferView}} |bufferView| and {{MLOperandDescriptor}} |descriptor|, run the following steps:
+    To <dfn>validate buffer with descriptor</dfn> given {{AllowSharedBufferSource}} |bufferSource| and {{MLOperandDescriptor}} |descriptor|, run the following steps:
+  </summary>
+    1. If |bufferSource|'s [=BufferSource/byte length=] is not equal to |descriptor|'s [=MLOperandDescriptor/byte length=] return false.
+    1. Switch on |bufferSource|:
+        <dl class=switch>
+            : {{ArrayBuffer}}
+            :: Return true.
+            : {{SharedArrayBuffer}}
+            :: Return true.
+            : {{ArrayBufferView}}
+            :: If |bufferSource|'s [=element type=] matches |descriptor|'s {{MLOperandDescriptor/dataType}} according to [this table](#appendices-mloperanddatatype-arraybufferview-compatibility) return true, otherwise return false.
+        </dl>
+</details>
+
+<details open algorithm>
+  <summary>
+    To <dfn>validate tensors with descriptors</dfn> given an {{MLNamedTensors}} |namedTensors| with [=record=]&lt;{{USVString}}, {{MLOperandDescriptor}}&gt; |namedDescriptors|:
   </summary>
-    1. If |bufferView|'s [=element type=] does not match to |descriptor|.{{MLOperandDescriptor/dataType}} according to [this table](#appendices-mloperanddatatype-arraybufferview-compatibility), return false.
-    1. If |bufferView|.\[[ByteLength]] is not equal to |descriptor|'s [=MLOperandDescriptor/byte length=], return false.
+    1. If |namedTensors|'s [=map/size=] is not equal to |namedDescriptors|'s [=map/size=], then return false.
+    1. [=map/For each=] |name| → |tensor| of |namedTensors|:
+        1. If |namedDescriptors|[|name|] does not [=map/exist=], then return false.
+        1. If |tensor|.{{MLTensor/[[descriptor]]}} is not equal to |namedDescriptors|[|name|], then return false.
```

Comment:
I think we can defer to a follow-up, but this is assuming an equality for maps. Ideally we'd either have a definition for that (https://garykac.github.io/procspec/#conditions) or equate the specific values.

This is not new to this addition - we use list equality all over the place, and that's not rigorously defined.